Automattic/wp-calypso
# Marketplace: Add reviews button to themes

Related to https://github.com/Automattic/dotcom-forge/issues/4557

|Reviews form | Successful message|
|-------|------|
|    ![CleanShot 2023-11-28 at 16 36 39@2x](https://github.com/Automattic/wp-calypso/assets/3519124/7c4aacf0-600e-4784-ac69-ebf0f26235b8)   |         ![CleanShot 2023-11-28 at 16 37 30@2x](https://github.com/Automattic/wp-calypso/assets/3519124/511bb8d0-b903-4830-a0d5-fa15098353ba)     |

## Proposed Changes

* Add a new button to the Theme details page to add a review
* Add a new modal with two fields to add a review to a theme
* Add functionality to send the review using the new review hooks
* Display success or error depending on the response of the endpoint call

## Testing Instructions



* Navigate to a theme details page, for example `/themes/twentytwentythree/:siteId`
* Make sure the flag `marketplace-add-review` is enabled by adding the `?flags=marketplace-add-review` to the url
* Check there is a button below the main header that reads `Add review`
* Click the button
* Make sure a modal is displayed with a form to add a review
* Open the Developer Tools and select the Network Panel to inspect the HTTP calls
* Populate the details and click on `Add new review`
* Make sure a call to the new endpoint `https://public-api.wordpress.com/wp/v2/sites/marketplace.wordpress.com/comments` is done with the expected payload
* The system currently returns an error. Make sure the error is displayed in the modal.

- Disable the feature flag adding the following to the URL `?flags=-marketplace-add-review`
- Make sure the `Add review` button is not displayed in the Theme details screen


## Pre-merge Checklist
